### ISSUE TITLE
TARS - Use of loss weights for "regular" text classification?

### ISSUE THREAD

**Issue by robinvanschaik** [flairNLP/flair#1930]
Hi all, 

Thanks for the great work on Flair so far.
I get stellar performance on an imbalanced dataset on the regular TextClassifier by using a loss weights dictionary as described [here](https://github.com/flairNLP/flair/issues/1897#issuecomment-708967367_).

I am really curious to try out the TARS classifier. The [tutorial](https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_10_TRAINING_ZERO_SHOT_MODEL.md) mentions that you can use the TARS classifier as a regular text classifier. 

However, this approach of passing a loss weight dictionary does not work with the TARSClassifier, and I bet that this would degrade performance in comparison to my old model. 
```
# Does not work. 
classifier =  TARSClassifier(document_embeddings,
                                 label_dictionary=label_dict,
                                 loss_weights=weight_dict )
```

Obviously it's due to an unexpected argument.

https://github.com/flairNLP/flair/blob/c83d771c0394d17e19183b819432d0e9aa1ed6a6/flair/models/text_classification_model.py#L560-L565

However, the TARS model calls a text-classifier object with the loss_weights explicitly set to None. 
https://github.com/flairNLP/flair/blob/c83d771c0394d17e19183b819432d0e9aa1ed6a6/flair/models/text_classification_model.py#L577-L582

Would it make sense to include the loss weights during training on the "regular" text classification task? 
Or does it not make sense by the way training is handled to enable zero-shot prediction?

Kind regards,

**Comment by kishaloyhalder** [comment]
Hi @robinvanschaik,

Thanks for trying out the tutorial. I understand your confusion with the `loss_weights`. Currently TARS does not support it. However, it has [shown](https://kishaloyhalder.github.io/pdfs/tars_coling2020.pdf) to be quite effective for long tailed classes in our zero/few-shot experiments. Maybe you would be interested to validate this hypothesis on your data. Feel free to share your observations. We would be interested to learn from you and will try to improve TARS going forward. Hope this helps!

with regards,
Kishaloy

**Comment by robinvanschaik** [comment]
Hi @kishaloyhalder, 

Thanks for the quick response. I will definitely check out the paper to better understand what the mechanics are behind the TARSClassifier. 

Regarding the `loss_weight`, I had a hunch but it is nice to have it confirmed. 👍 

The reason why I am so interested in this zero-shot approach is that I am currently dealing with categorising incoming requests through a contact form. 
However, my training dataset is labeled by humans, which means that categories often get conflated, are limited in size or are noisy. As such, a number of categories are dropped from the training set.

My hypotheses would be that that A) that zero-shot classification could mean that we can apply the model to the dropped categories as well, and B) whether it will possible to categorise incoming requests in another domain.

**Comment by kishaloyhalder** [comment]
Sounds good @robinvanschaik! The model is meant for dynamic situations around classification datasets like yours. I suggest you try out training the model as shown [here](https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_10_TRAINING_ZERO_SHOT_MODEL.md#training-a-tars-classification-model). Once you have trained a model on your data, you can play around with different sets of labels as shown [here](https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_10_TRAINING_ZERO_SHOT_MODEL.md#use-case-2-as-a-zero-shot-predictor-on-arbitrary-ad-hoc-set-of-labels) and [here](https://github.com/flairNLP/flair/blob/master/resources/docs/TUTORIAL_10_TRAINING_ZERO_SHOT_MODEL.md#use-case-3-as-a-zero-shot-predictor-on-an-unseen-task). Feel free to reach out if you have any other question.

with regards,
Kishaloy
